package/MDAnalysis/coordinates/base.py

```diff
@@ -1293,13 +1294,21 @@ def check_slice_indices(self, start, stop, step):
             range(10)[10:-1:-1]  # []
 
         """
-        for var, varname in (
-                (start, 'start'),
-                (stop, 'stop'),
-                (step, 'step')
-        ):
-            if not (isinstance(var, int) or (var is None)):
+
+        slice_dict = {'start': start, 'stop': stop, 'step': step}
+        for varname in slice_dict:
```

Comment:
`for varname, var in six.iteritems(slice_dict):` will result in py2/3 compatible code that always uses generators.